### PR TITLE
docs: Incorrect rule type in thanos rules federation proposal

### PR DESCRIPTION
**Pull request by benjaminhuo**
Signed-off-by: Benjamin <benjamin@yunify.com>

* [x] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes
- Correct incorrect rule type

## Verification
N/A

### CODE REVIEW

**Comment by benjaminhuo** [comment]
@bwplotka , the incoming rule group in scenario 1 looks like below, where rule r1 in group a has type alert:
```
group: a
   recording:<name:"a1" last_evaluation:<seconds:1 > >
   alert:    <name:"r1" last_evaluation:<seconds:1 > >
group: b
   recording:<name:"r1" last_evaluation:<seconds:1 > >
group: a
   recording:<name:"r2" last_evaluation:<seconds:1 > >
```
But the output looks like this, where rule r1 in group a has type recording:
```
group: a
   alert:    <name:"a1" last_evaluation:<seconds:1 > >
   recording:<name:"r1" last_evaluation:<seconds:1 > >
   recording:<name:"r2" last_evaluation:<seconds:1 > >
group: b
   recording:<name:"r1" last_evaluation:<seconds:1 > >
```

And there is one comments after that saying `Note in the example above how the recording rule r1 is not deduplicated as it is contained in two different groups.` ,  I think it make sense for rule r1 in group a having type `recording` instead of alert, because rules with different type shouldn't be deduplicated.

image

**Comment by benjaminhuo** [comment]
@bwplotka , the incoming rule group in scenario 1 looks like below, where rule `a1` in group `a` has type `recording` and rule `r1` in group `a` has type `alert`:
```
group: a
   recording:<name:"a1" last_evaluation:<seconds:1 > >
   alert:    <name:"r1" last_evaluation:<seconds:1 > >
group: b
   recording:<name:"r1" last_evaluation:<seconds:1 > >
group: a
   recording:<name:"r2" last_evaluation:<seconds:1 > >
```
But the output looks like this, where rule `a1` in group `a` has type `alert` and rule `r1` in group `a` has type `recording`:
```
group: a
   alert:    <name:"a1" last_evaluation:<seconds:1 > >
   recording:<name:"r1" last_evaluation:<seconds:1 > >
   recording:<name:"r2" last_evaluation:<seconds:1 > >
group: b
   recording:<name:"r1" last_evaluation:<seconds:1 > >
```

image

File: docs/proposals/202003_thanos_rules_federation.md [hunk 1]
```diff
@@ -88,8 +88,8 @@ As specified, the rule type and then rule name is used for deduplication.
 Given the following stream of incoming rule groups and containing recording/alerting rules:
 ```text
 group: a
-   recording:<name:"a1" last_evaluation:<seconds:1 > >
-   alert:    <name:"r1" last_evaluation:<seconds:1 > >
+   alert:    <name:"a1" last_evaluation:<seconds:1 > >
```
[the remark this answers is not in the document]
Comment (s-urbaniak) [reply docs/proposals/202003_thanos_rules_federation.md:91]: ah i see, good catch! I mixed up the types and names :facepalm: 

However, this scenario also outlines the ordering of rule types (alerts come before recording rules). Therefore do you mind to keep the ordering in the incoming stream, where the recording rule arrives before the alert but is sorted such that the alert comes before the recording rule?

so this incoming rules are:
```
   recording:<name:"r1" last_evaluation:<seconds:1 > >
   alert:    <name:"a1" last_evaluation:<seconds:1 > >
```

[the remark this answers is not in the document]
Comment (benjaminhuo) [reply docs/proposals/202003_thanos_rules_federation.md]: @s-urbaniak Done

**Comment by s-urbaniak** [comment]
lgtm :+1: thank you!

File: docs/proposals/202003_thanos_rules_federation.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bwplotka) [reply docs/proposals/202003_thanos_rules_federation.md]: ah =D

**Review by bwplotka — APPROVED** [review]
Thanks!

**Pull request merged**